### DISCUSSION TITLE
White Screen on Dynamic Routes in Vercel Production

### DISCUSSION THREAD

**Discussion by willfeldman** [vercel/community#5293] [Help]
### Summary

**Issue Summary:**
When running my React application locally using vercel dev, everything functions correctly. However, after deploying to Vercel, navigating to any non-root dynamic route (e.g., /experience/:id) results in only a white screen being displayed. The expected JavaScript functionality isn't operational, leading to a non-interactive page. This issue only occurs in the production environment on Vercel and not during local development.

**Details:**
- The problem seems to be specific to routes with dynamic segments (e.g., /experience/:id).
- The index.html is being served, but the associated JavaScript isn't loading as expected.
- There's a suspicion that HTML is not being processed correctly, possibly being treated as a script, which might be causing the problem.
- The application uses React Router for client-side routing. I am not using the pages infrastructure for routing. 

**Troubleshooting Already Done:**
- Confirmed that routing configuration in App.tsx appears correct for React Router.
- Verified that the issue is not present when running the application locally.
- The issue persists across all my dynamic routes in the production environment.
- I have tried changing the vercel.json a bunch of time but with no luck...

### Example

https://willfeldman.com/experience/6

### Steps to Reproduce

The code is here:
https://github.com/willfeldman/Portfolio

Everything works with `vercel dev`, but when deployed and going to the dynamic routes, a white screen is shown. Go to https://willfeldman.com/experience/6 to see the issue in action. 

Routing info is in the App.tsx (https://github.com/willfeldman/Portfolio/blob/main/src/App.tsx)

**Comment by krishnaacharyaa** [comment]
Hi @willfeldman, I see in your `vercel.json` there are no redirects in place, can you try adding this

```
{
  "rewrites": [{ "source": "/(.*)", "destination": "/" }]
}
```

p.s: You need to set vercel.json to have these rewrites for the dynamic routes to work it's not just `employee/:id` which is not working if you carefully see, even `/image` is not working `https://willfeldman.com/image`.

**Reply by willfeldman** [reply]
@krishnaacharyaa I made the update, but the issue persists. I've tried several similar changes to `vercel.json` but nothing has worked :/ I think it might be something related to a JavaScript file being read as HTML but I'm not sure. Thank you for the reply though! Do you happen to have any other ideas?

**Reply by krishnaacharyaa** [reply]
> JavaScript file being read as HTML

Can you please elaborate on this, I didn't get you...

**Comment by krishnaacharyaa** [comment]
@willfeldman Hey did it solve? or you want me to solve it ?

**Comment by eRuaro** [comment]
Same issue here even with having a vercel.json
